### ISSUE TITLE
Placement issue using align 'center' and data-ss-colspan > 1

### ISSUE THREAD

**Issue by pascalbayer** [AshesOfOwls/jquery.shapeshift#49]
The issue only occurs using align 'center' and data-ss-colspan > 1.
The inner div's floating out on the right side of the container and elements are incorrectly centered in the container. It is somehow related to the calculations in:

``` javascript
Plugin.prototype.setGridColumns //inside this function
columns = options.columns || Math.floor((inner_width + gutter_x) / col_width);
grid_width = (columns * col_width) - gutter_x; return globals.child_offset += (inner_width - grid_width) / 2;
```

An example is the original jsfiddle sample [http://jsfiddle.net/McPants/LNysC/light/](http://jsfiddle.net/McPants/LNysC/light/) when resizing before last container is moved to the next row.

**Comment by pascalbayer** [comment]
![image](https://f.cloud.github.com/assets/3659792/612549/e4644f3a-cdee-11e2-9b9b-429f48bb297f.png)

**Comment by AshesOfOwls** [comment]
Ah yeah it seems that when calculating the offset it isn't accounting for multiwidth items. Looking into it right now.

**Comment by AshesOfOwls** [comment]
Found out what was wrong

``` coffeescript
# Columns cannot exceed children
children_count = @parsedChildren.length
if columns > children_count
   columns = children_count
```

This piece of code was to make sure that there weren't more columns than there were children. If there were less children than there could be columns, it will set the amount of columns to the total number of children. The problem is that if there are multiwidth children then it is ignoring the fact they are multiwidth. Will be uploading the fix for this very soon.

**Comment by pascalbayer** [comment]
The alignment works now, but only when there is no row break, after a row break it's floating out now on the left side.

**Comment by AshesOfOwls** [comment]
Hmm ok I'll check it out again

**Comment by qbizns** [comment]
Hi
 Thanks for great plugin 
i try to use it with 960 grid system every thing work perfect except if i put ( 8col + 3col + 1col ) and ( 9col + 3col ) i use 12 col system 
and i set width to

.container > div[data-ss-colspan="1"] { width: 60px;  }
.container > div[data-ss-colspan="2"] {  width: 140px;  }
.container > div[data-ss-colspan="3"] {  width: 300px;  }
.container > div[data-ss-colspan="4"] {  width: 220px;  }
.container > div[data-ss-colspan="5"] {  width: 380px;  }
.container > div[data-ss-colspan="6"]{  width: 460px;  }
.container > div[data-ss-colspan="7"] {  width: 540px;  }
.container > div[data-ss-colspan="8"] {  width: 620px;  }
.container > div[data-ss-colspan="9"] {  width: 700px;  }
.container > div[data-ss-colspan="10"] {  width: 780px;  }
.container > div[data-ss-colspan="11"] {  width: 860px;  }
.container > div[data-ss-colspan="12"] {  width: 940px;  }

**Comment by AshesOfOwls** [comment]
@pascalbayer You are right with it still not working on multiple rows, I am looking at it now

**Comment by AshesOfOwls** [comment]
@qbizns Please let me know if this fix is also for your problem. If not then I am not quite sure whats wrong, if you could explain what goes wrong a little more in depth it would help me a lot, or even a live demo.

**Comment by pascalbayer** [comment]
Thanks a lot for your fantastic work! Everything seems to work fine now!

**Comment by AshesOfOwls** [comment]
For sure, thank you for the help in troubleshooting it.

**Comment by pascalbayer** [comment]
After playing around for a while I've found another 'bug'. The grid is only centered depending on the first row, what will cause floating out elements on the right side in the second row when it's elements are larger than the ones in the first row.
![image](https://f.cloud.github.com/assets/3659792/634543/96e59cf0-d21c-11e2-9090-d4229357b8a9.png)
So it maybe would be better to center each row individually, to have a real center alignment.

**Comment by AshesOfOwls** [comment]
Yeah I would really like to put this in but it gets complicated once there are elements that are differing heights. I am trying to keep the plugin as generic as possible but it may be necessary to implement different alignment algorithms based on if your grid allows heights that vary.

I can add this as a potential 3.0 feature but I already have a bunch on my list for now. I'll reopen this and tag it for 3.0 though and keep you updated if I get a chance to implement it. If not it may end up being 3.x or 4.0

**Comment by bufke** [comment]
Related to this - elements go off the screen on very small viewports, such as phone sized. Try viewing the original jsfiddle and make the viewport very small to see.

If the developer was using responsive design, the elements width could be made smaller using a media query. However there is no simple way to change the data-ss-colspan. 

Here is an example of what I mean.
http://imgur.com/sWMslpE

A simple fix might be a sanity check to never allow a left of < 0. I don't see any reason this would ever be desired.

**Comment by AshesOfOwls** [comment]
I am currently working on the rough edges regarding this and other basic positioning related stuff in 3.0. I'll also be trying to allow for percentage based elements too. Hopefully it will be out within a month but I'm not real sure if thats feasible or not. Thanks for the heads up though @bufke

**Comment by noctivityinc** [comment]
Im having the same issue.  Any thoughts on when 3.0 will be ready or if there is a fix for this?
